robottelo/utils/__init__.py

```diff
@@ -80,3 +91,178 @@ def slugify_component(string, keep_hyphens=True):
+        consumer_export_zip = zipfile.ZipFile(io.BytesIO(template_zip.read('consumer_export.zip')))
+
+        # Generate a new consumer_export.zip file changing the consumer
+        # uuid.
+        consumer_export = io.BytesIO()
+        with zipfile.ZipFile(consumer_export, 'w') as new_consumer_export_zip:
+            for name in consumer_export_zip.namelist():
+                if name == 'export/consumer.json':
+                    consumer_data = json.loads(consumer_export_zip.read(name).decode('utf-8'))
+                    consumer_data['uuid'] = str(uuid.uuid1())
+                    if org_environment_access:
+                        consumer_data['contentAccessMode'] = 'org_environment'
+                        consumer_data['owner'][
+                            'contentAccessModeList'
+                        ] = 'entitlement,org_environment'
+                    new_consumer_export_zip.writestr(name, json.dumps(consumer_data))
+                else:
+                    new_consumer_export_zip.writestr(name, consumer_export_zip.read(name))
+
+        # Generate a new manifest.zip file with the generated
+        # consumer_export.zip and new signature.
+        manifest = io.BytesIO()
+        with zipfile.ZipFile(manifest, 'w', zipfile.ZIP_DEFLATED) as manifest_zip:
+            consumer_export.seek(0)
+            manifest_zip.writestr('consumer_export.zip', consumer_export.read())
+            consumer_export.seek(0)
+            signature = self.private_key.sign(
+                consumer_export.read(), padding.PKCS1v15(), hashes.SHA256()
+            )
+            manifest_zip.writestr('signature', signature)
+        # Make sure that the file-like object is at the beginning and
+        # ready to be read.
+        manifest.seek(0)
+        return manifest
+
+    def original(self, name='default'):
+        """Returns the original manifest as a file-like object.
+
+        :param name: A name of the manifest as defined in robottelo.properties
+
+        Be aware that using the original manifest and not removing it
+        afterwards will make it impossible to import it to any other
+        Organization.
+
+        Make sure to close the returned file-like object in order to clean up
+        the memory used to store it.
+        """
+        if self.signing_key is None or self.template is None or self.template.get(name) is None:
+            self._download_manifest_info(name)
+        return io.BytesIO(self.template[name])
+
+
+# Cache the ManifestCloner in order to avoid downloading the manifest template
+# every single time.
+_manifest_cloner = ManifestCloner()
+
+
+class Manifest:
+    """Class that holds the contents of a manifest with a generated filename
+    based on ``time.time``.
+
+    To ensure that the manifest content is closed use this class as a context
+    manager with the ``with`` statement::
+
+        with Manifest() as manifest:
+            # my fancy stuff
```

Comment:
Is this relevant :)

Reply:
@adarshdubey-star It is done as an example to use  it.